src/ucp/rndv/proto_rndv.c

```diff
@@ -558,14 +566,27 @@ static size_t ucp_proto_rndv_ats_pack_ack(void *dest, void *arg)
     return ucp_proto_rndv_pack_ack(req, dest, req->send.state.dt_iter.length);
 }
 
+static UCS_F_ALWAYS_INLINE ucs_status_t
+ucp_proto_rndv_ats_complete(ucp_request_t *req)
+{
+    ucp_request_t *rreq = ucp_request_get_super(req);
+
+    if (req->send.rndv.rkey != NULL) {
+        ucp_proto_rndv_rkey_destroy(req);
+    }
+
+    ucp_datatype_iter_cleanup(&req->send.state.dt_iter, UCP_DT_MASK_ALL);
+    return ucp_proto_rndv_recv_complete_status(req, rreq->status);
+}
+
 ucs_status_t ucp_proto_rndv_ats_progress(uct_pending_req_t *uct_req)
 {
     ucp_request_t *req = ucs_container_of(uct_req, ucp_request_t, send.uct);
 
     return ucp_proto_rndv_ack_progress(req, req->send.proto_config->priv,
                                        UCP_AM_ID_RNDV_ATS,
                                        ucp_proto_rndv_ats_pack_ack,
-                                       ucp_proto_rndv_recv_complete);
+                                       ucp_proto_rndv_ats_complete);
```

Comment:
why need to change this? now there is extra branch (and call to ucp_datatype_iter_cleanup) in rndv/get protocols as well
we probably need to have special "ucp_proto_rndv_ats_progress" replacement only for rndv/ats protocol - that means, fully implement ucp_proto_rndv_ats_proto_progress function without calling ucp_proto_rndv_ats_progress

Reply:
Hmm.. gtest passes rndv tests with origin code, so just reverting for now